### ISSUE TITLE
summaryToUrlTree only returns a tree from the first header it encounters in md file

### ISSUE THREAD

**Issue by dylviz** [Open-EdTech/next-mdx-books#7]
In a md with multiple headers, only the first list are iterated. For example, as defined in the GitBook [docs](https://docs.gitbook.com/product-tour/git-sync/content-configuration), a summary md will have the following structure:

```
‌# Summary​

## Use headings to create page groups like this one​

* [First page's title](page1/README.md)    
    * [Some child page](page1/page1-1.md)    
    * [Some other child page](part1/page1-2.md)

* [Second page's title](page2/README.md)    
    * [Some child page](page2/page2-1.md)    
    * [Some other child page](part2/page2-2.md)    

## A second-page group​

* [Yet another page](another-page.md)
``` 
But this will stop parsing when it reaches "A second-page group".

**Comment by MatthewCaseres** [comment]
Thanks for raising this, seems to be a duplicate of the other open issue. Will close the other issue in favor of this one.

**Comment by dylviz** [comment]
Oops missed that. Thanks for letting me know. I've been looking into this for a few hours, but can't find what's causing this yet. I'll update when I find something

**Comment by MatthewCaseres** [comment]
Cool, i haven't looked at this project for a long time. The new [GitHub code viewer](https://docs.github.com/en/repositories/working-with-files/managing-files/navigating-files-with-the-new-code-view) (I think it is in beta) provides sidebar navigation for GitHub repos. So that is probably what I would use nowadays. Going to archive the repo just so people don't get the impression that it is actively maintained.

**Comment by dylviz** [comment]
If you're interested, I think a found a workable fix by just returning an array of trees for summaryToUrlTree. I'm not great with ts so I would appreciate a review on the PR I just submitted, but if it's preferred I can just maintain updates on my personal fork. #8

**Comment by MatthewCaseres** [comment]
Yeah I will archive this repo and point to your fork as a fork that is maintained under the assumption that you would be more able to help someone with the codebase if they have an interest.

Definitely recommend adding tests to the code. I was pretty new to development when I wrote this.

**Comment by dylviz** [comment]
Ok, sure thing. Thanks!
